jbpm-flow-builder/src/main/java/org/jbpm/compiler/ProcessBuilderImpl.java

```diff
@@ -426,7 +438,7 @@ private String createSplitRule(Process process,
         		connection.getToType() + "\"  @Propagation(EAGER) \n" +
         	"      ruleflow-group \"DROOLS_SYSTEM\" \n" +
         	"    when \n" +
-        	"      " + constraint + "\n" +
+        	"      " + constraint + getEntryPoint(constraint, process) +  "\n" +
```

Comment:
Same concern around backwards compatibility, doesn't this also make it difficult to actually try and share data across instances, which might be useful in some cases as well.  Or are we going to consider this an anti-pattern since this will result in different behaviour depending on session strategy?